File: BusinessNetwork-Extension/insurancebusinessnetwork/contracts/src/test/java/net/corda/samples/businessmembership/contracts/StateTests.java
package net.corda.samples.businessmembership.contracts;

import net.corda.core.identity.Party;
import net.corda.samples.businessmembership.states.InsuranceState;
import org.junit.Test;

public class StateTests {

    @Test
    public void hasFieldOfCorrectType() throws NoSuchFieldException {
        // Does the message field exist?
        InsuranceState.class.getDeclaredField("insurer");
        // Is the message field of the correct type?
        assert(InsuranceState.class.getDeclaredField("insurer").getType().equals(Party.class));
    }
}

File: kotlin-samples/BusinessNetworks/insurancebusinessnetwork/contracts/src/test/kotlin/net/corda/samples/businessmembership/contracts/StateTests.kt
package net.corda.samples.businessmembership.contracts

import net.corda.core.identity.Party
import net.corda.samples.businessmembership.states.InsuranceState
import net.corda.testing.node.MockServices
import org.junit.Assert
import org.junit.Test

class StateTests {
    private val ledgerServices = MockServices()

    @Test
    fun hasFieldOfCorrectType() {
        // Does the message field exist?
        InsuranceState::class.java.getDeclaredField("insurer")
        // Is the message field of the correct type?
        Assert.assertEquals(InsuranceState::class.java.getDeclaredField("insurer").type, Party::class.java)
    }
}